src/playlist.js

```diff
@@ -45,20 +63,10 @@ export const liveEdgeDelay = (master, media) => {
   const partsAndSegments = getPartsAndSegments(media);
   const hasParts = partsAndSegments.length &&
     typeof partsAndSegments[partsAndSegments.length - 1].partIndex === 'number';
-
-  // get the last three part/segment durations
-  let lastThreeDurations = 0;
-
-  if (partsAndSegments.length >= 3) {
-    for (let i = partsAndSegments.length - 1; i > partsAndSegments.length - 4; i--) {
-      // segment missing a duration, we cannot calculate
-      if (!partsAndSegments[i].duration) {
-        lastThreeDurations = 0;
-        break;
-      }
-      lastThreeDurations += partsAndSegments[i].duration;
-    }
-  }
+  // by default we use the last three durations of segments if
+  // part target duration or target duration isn't found.
+  // see: https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-08#section-4.4.3.8
```

Comment:
I may be missing something, but I don't think that section of the spec, if referring to `HOLD-BACK`, matches the comment here, since that is saying to use part target duration or target duration when the parameter is not available, and doesn't refer to the use of segment durations. Do we just have this in case the required target durations are missing?

Reply:
Yes we use three segment durations in case target durations are not available for whatever reason.

Reply:
I'm wondering if it's worth it to add the logic for this here if those are required attributes. It adds logic to the code that we shouldn't really encounter, and if we were to, then it's something that should be resolved by the creator of the stream, since target duration is always required (https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-08#section-4.4.3.1) and part target duration is required when parts are used (https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-08#section-4.4.3.7).

I think if we're trying to account for missing target durations we should do that at a higher level, when we first parse the manifest. It would help to simplify the code.

Reply:
ok I will remove the last three durations logic.

Reply:
What about just default to 10s target duration in case somehow it isn't set or whatever? While I agree that bad streams are bad, we should still try to play as much as possible and having a default, if bad value, is worth it. Plus, it should make things simpler than trying to figure out segment duration and using that for target duation.